tachyon/c/zk/plonk/halo2/bn254_prover_unittest.cc

```diff
@@ -276,6 +276,132 @@ TEST_P(ProverTest, CommitLagrange) {
   EXPECT_EQ(c::base::native_cast(*point), expected);
 }
 
+TEST_P(ProverTest, BatchCommit) {
+  using Poly =
+      math::UnivariateDensePolynomial<math::bn254::Fr, c::math::kMaxDegree>;
+  std::vector<Poly> polys =
+      base::CreateVector(4, []() { return Poly::Random(5); });
+
+  tachyon_halo2_bn254_prover_batch_start(prover_, polys.size());
+  for (size_t i = 0; i < polys.size(); ++i) {
+    tachyon_halo2_bn254_prover_batch_commit(prover_, c::base::c_cast(&polys[i]),
+                                            i);
+  }
+
+  std::vector<math::bn254::G1AffinePoint> points(polys.size());
+  tachyon_halo2_bn254_prover_batch_end(
+      prover_,
+      const_cast<tachyon_bn254_g1_affine*>(c::base::c_cast(points.data())),
+      points.size());
+
+  for (size_t i = 0; i < points.size(); ++i) {
+    const Poly& poly = polys[i];
+    math::bn254::G1AffinePoint expected;
+    switch (static_cast<zk::plonk::halo2::PCSType>(prover_->pcs_type)) {
```

Comment:
For all instances of `zk::plonk::halo2::` (excluding instances of `c::zk::plonk::halo2::`)
```suggestion
    switch (static_cast<PCSType>(prover_->pcs_type)) {